flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -1437,36 +1437,30 @@ object AggregateUtil {
 
           case collect: SqlAggFunction if collect.getKind == SqlKind.COLLECT =>
             aggregates(index) = new CollectAggFunction(FlinkTypeFactory.toTypeInfo(relDataType))
-            accTypes(index) = aggregates(index).getAccumulatorType
 
           case udagg: AggSqlFunction =>
             aggregates(index) = udagg.getFunction
-            accTypes(index) = udagg.accType
 
           case unSupported: SqlAggFunction =>
             throw new TableException(s"Unsupported Function: '${unSupported.getName}'")
         }
       }
+      accTypes(index) = getAccumulatorTypeOfAggregateFunction(aggregates(index))
     }
 
     val accSpecs = new Array[Seq[DataViewSpec[_]]](aggregateCalls.size)
 
     // create accumulator type information for every aggregate function
     aggregates.zipWithIndex.foreach { case (agg, index) =>
```

Comment:
I dont think changes to this logic is needed if you follow the above comment.

Reply:
Without this change, we need to add `accTypes(index) = aggregates(index).getAccumulatorType` for every built-in agg which uses `DataView` in the future. I think this change makes the logical more simple and clear.

Reply:
+1 you are right. I missed the 4 lines above. 👍

Reply:
+1